src/plane.ts

```
@@ -9,19 +9,104 @@
+  };
+  const interrupts = [toControls.yaw, toControls.pitch, toControls.roll];
+  const controlLoop = {
+    frequency: 600,
+    interrupts: interrupts,
+  };
+  const controlLoops = [controlLoop];
+  const plane = {
+    apiVersion: 0.1,
+    id: 0,
+    author: 'default',
+    created: '2024-02-27T05:27:00.000Z',
+    updated: '2024-02-27T05:27:00.000Z',
+    basicGeometry: basicGeometry,
+    controlLoops: controlLoops,
+  };
+  return plane;
+}
 
 export class Plane {
   private static readonly MATERIAL = new THREE.MeshNormalMaterial();
-  private wingspan: number;
-  private length: number;
-  private chord: number;
-  private model: THREE.Group;
-
-  constructor(wingspan: number, length: number, chord: number) {
-    this.wingspan = wingspan;
-    this.length = length;
-    this.chord = chord;
-    this.model = this.genStandinGeometry(wingspan, length, chord);
+  private static readonly defaultWingspan = 1.1;
+  private static readonly defaultLength = 0.4;
+  private static readonly defaultChord = 0.1;
+  model: THREE.Group;
+  private planeSpecs: planeSpecs;
+
+  constructor() {
+    this.planeSpecs = option.getOrElse(() => genBasicPlane())(
+      this.importPlane(JSON.stringify(defaultPlane))
+    );
+    if (this.planeSpecs.basicGeometry) {
+      this.model = this.genStandinGeometry(
+        this.planeSpecs.basicGeometry?.wingspan,
+        this.planeSpecs.basicGeometry?.length,
+        this.planeSpecs.basicGeometry?.chord
+      );
+    } else {
+      this.model = this.genStandinGeometry(
+        Plane.defaultWingspan,
+        Plane.defaultLength,
+        Plane.defaultChord
+      );
+    }
+  }
+
+  importPlane(specsJson: string): option.Option<planeSpecs> {
+    let planeSpecsIn;
+    try {
+      planeSpecsIn = JSON.parse(specsJson);
+    } catch (err) {
+      console.log(`JSON parse error on plane import. Specific error: ${err}`);
+      return option.none;
+    }
+    if (planeSpecsIn.basicGeometry || planeSpecsIn.vertexGeometry)
```

This should be XOR -- `!!planeSpecsIn.basicGeometry ^ !!planeSpecsIn.vertexGeometry`

It's better to fail loudly when APIs aren't followed to ensure people don't misinterpret what the app is doing and correct their inputs. Similarly, it's probably best not to fall back to the output of `genBasicPlane()` if you expect people to set up the `default-plane.json` correctly b/c people might think it's working when it's not.

---

For now I'm just yelling at them with an alert.